### PR TITLE
[Fixes #354] fixes CubicFermionicSimulationGate._value_equality_values_

### LINKED ISSUE #354: CubicFermionicSimulationGate is not actually periodic in its parameters
This is assumed by its `__eq__` method. For instance, in
```
import numpy as np
import cirq
import openfermioncirq as ofc


G = ofc.CubicFermionicSimulationGate((1, 1, 1))
H = ofc.CubicFermionicSimulationGate((1 + 2 * np.pi, 1, 1))

U = cirq.unitary(G)
V = cirq.unitary(H)

assert G == H
cirq.testing.assert_allclose_up_to_global_phase(U, V, atol=1e-5)
```
the first assertion passes but not the second. It turns out that the definition of the gate does not imply periodicity; my bad for letting it pass review. @bryano

### PR DESCRIPTION
**Pull request by bryano**

### CODE REVIEW

**Comment by bryano** [comment]
The gate actually is periodic, but only when all the weights are shifted by the same amount. This quick fix avoids false positives.

File: openfermioncirq/gates/fermionic_simulation.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def _eigen_components(self):
         return components
 
     def _value_equality_values_(self):
-        return tuple(_canonicalize_weight(w * self.exponent)
+        return tuple((w * self.exponent,)
                 for w in list(self.weights) + [self._global_shift])
```
Comment (kevinsung) [inline openfermioncirq/gates/fermionic_simulation.py]: Why can't the definition simply be
```
tuple(list(self.weights) + [self.exponent, self._global_shift])
```
?

Comment (bryano) [reply openfermioncirq/gates/fermionic_simulation.py]: What if the exponent is zero?

Comment (kevinsung) [reply openfermioncirq/gates/fermionic_simulation.py]: I didn't think of that.

**Pull request merged**